packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -1696,17 +1681,22 @@ https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scope
       claimAndMint: async (
         derivedKeyId: BytesLike,
         signatures: IPubkeyRouter.SignatureStruct[],
-        txOpts?: any
+        txOpts: ethers.PayableOverrides & { from?: string } = {}
       ) => {
         try {
-          const cost = await this.pkpNftContract.read.mintCost();
-          const tx = await this.pkpNftContract.write.claimAndMint(
-            2,
-            derivedKeyId,
-            signatures,
-            txOpts ?? { value: cost }
+          if (!txOpts.value) {
```

Comment:
Would prefer if we didn't mutate `txOpts` since it could have been passed in, and setting the `value` could result in an unexpected side-effect if those opts were re-used for another call by the caller.

Reply:
On this particular case: This is also fixing the issue where the transaction would fail when user passes something for `txOpts` but without `cost` in it. The function would calculate it but not pass it to the tx, that required it

Anyway, I get your point and added an intermediate variable for all that